ci.py

```diff
@@ -74,6 +93,25 @@ def setup_pants_version(test_pants_version: PantsVersion):
     write_config(original_config)
 
 
+@contextmanager
+def setup_python_version(test_python_version: PythonVersion):
+  """Modify pants.ini to allow the Python version to be unspecified or change to what was requested."""
+  original_config = read_config()
+  updated_config = read_config()
+  config_entry = "pants_runtime_python_version"
+  if test_python_version == PythonVersion.unspecified:
+    updated_config.remove_option(GLOBAL_SECTION, config_entry)
+  else:
+    updated_config[GLOBAL_SECTION][config_entry] = test_python_version.value
+  write_config(updated_config)
+  try:
+    yield
+  except subprocess.CalledProcessError:
```

Comment:
I'm not sure why this `except` is here; isn't the behaviour the same without it?